src/ethr-did-resolver.js

```diff
@@ -212,14 +212,11 @@ export default function register(conf = {}) {
     }
     return history
   }
-  async function resolve(did, parsed) {
+  return async function resolve(did, parsed) {
     if (!parsed.id.match(/^0x[0-9a-fA-F]{40}$/))
       throw new Error(`Not a valid ethr DID: ${did}`)
     const owner = await didReg.identityOwner(parsed.id)
     const history = await changeLog(parsed.id)
     return wrapDidDocument(did, owner['0'], history)
   }
-  registerMethod('ethr', resolve)
```

Comment:
If this line gets removed, the responsibility of registering the resolver to the 'ethr' key should be replaced somewhere else.  One possible pattern is by injecting the `did-resolver` dependency into `register`

Reply:
@mi-xu Im not sure I understand what you mean by this?